File: app/src/main/java/com/drivingschoolrwandaapp/database/entities/UserSubscriptionWithPlan.java
package com.drivingschoolrwandaapp.database.entities;

import androidx.room.Embedded;
import androidx.room.Relation;

public class UserSubscriptionWithPlan {
    @Embedded
    public UserSubscriptionEntity userSubscription;

    @Relation(
         parentColumn = "subscriptionPlanId",
         entityColumn = "id"
    )
    public SubscriptionPlan subscriptionPlan;
}

File: app/src/main/java/com/drivingschoolrwandaapp/database/entities/UserSubscriptionWithPlan.kt
package com.drivingschoolrwandaapp.database.entities

import androidx.room.Embedded
import androidx.room.Relation

data class UserSubscriptionWithPlan(
    @Embedded var userSubscription: UserSubscriptionEntity? = null,
    @Relation(
        parentColumn = "subscriptionPlanId",
        entityColumn = "id"
    )
    var subscriptionPlan: SubscriptionPlan? = null
)
